### PR TITLE
[CI] Add code coverage upload workflow

### PR DESCRIPTION
**Pull request by vettoreldaniele**
This workflow is responsible for retrieving code coverage report artifacts created by pull request CI workflows, uploading them to Google Cloud Storage, and leaving a comment on the originating PR with a link to the code coverage report.

The advantage of having a separate workflow is that failures of this workflow will not impact pull requests.

The workflow is set up so that it runs automatically after a pull-request CI workflow. Since it accesses repository's secrets, it's only configured to run in the main GpuOpen-Drivers/llpc repository.

Example run: https://github.com/vettoreldaniele/llpc/actions/runs/1684053472 that left a comment in PR https://github.com/vettoreldaniele/llpc/pull/5.

### CODE REVIEW

**Comment by vettoreldaniele** [comment]
This may not run until this PR is merged, but let's wait for checks to finish. I believe it's safe to merge once reviewed and approved, since even if it doesn't end up working, it will not impact current CI workflows, and I can fix it separately.

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

**Comment by vettoreldaniele** [comment]
PTAL @kuhar

File: .github/workflows/upload-coverage-report.yml
Comment (kuhar) [inline .github/workflows/upload-coverage-report.yml:18]: Would it be possible to create just one comment? IE instead of running this job in 2 configurations, run one job but loop over all reports?

If this is tricky to implement, it's fine as-is.

File: .github/workflows/upload-coverage-report.yml
Comment (kuhar) [inline .github/workflows/upload-coverage-report.yml:40]: Do these have access to our gcloud login? If yes, could we first download the artifacts and authenticate with gcloud later?

File: .github/workflows/upload-coverage-report.yml
Comment (kuhar) [inline .github/workflows/upload-coverage-report.yml:6]: Later in the file we specify `workflow: check-amdllpc-docker.yml`. Would it be possible to use the filename instead of the textual name here as well?

File: .github/workflows/upload-coverage-report.yml
Comment (kuhar) [inline .github/workflows/upload-coverage-report.yml:46]: So this won't overwrite the previous coverage reports, right? Are run ids unique within a repo?

File: .github/workflows/upload-coverage-report.yml
Comment (vettoreldaniele) [reply .github/workflows/upload-coverage-report.yml:18]: I thought about that but decided not to do it since there's no way to "loop" in GitHub Actions workflows, meaning some of the steps or workflow code would have to be manually repeated (for example the comment's contents or the download action). Since the job takes very little time, I prefer to leave as is until there's a more concrete need to change. This has the advantage that we can easily change the configurations in the matrix.

File: .github/workflows/upload-coverage-report.yml
Comment (vettoreldaniele) [reply .github/workflows/upload-coverage-report.yml:46]: Each workflow has an unique run ID. Since each workflow uploads its own coverage reports, this effectively means each coverage report artifact has an unique ID associated with it, and we are guaranteed that we create a unique folder for each report.

File: .github/workflows/upload-coverage-report.yml
Comment (vettoreldaniele) [reply .github/workflows/upload-coverage-report.yml:6]: No, this is a limitation of GitHub. The other action is a custom action that handles it specially.

File: .github/workflows/upload-coverage-report.yml
Comment (vettoreldaniele) [reply .github/workflows/upload-coverage-report.yml:40]: I will move these before gcloud authentication to avoid possible security issues.

**Comment by vettoreldaniele** [comment]
V1: move gcloud authentication as late as possible to avoid other actions possibly using gcloud credentials.

**Review by kuhar — APPROVED** [review]
LGTM. Thanks for clarifying the limitations.

**Comment by kuhar** [comment]
test this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 6b662c42cd5f6cfc5756fd4a3eaff4eb2eeeb39d
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 39f8940199e60c44d4211cf8165dfd12876316fa</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev bcec463efe5260776d486a5e3da0c549bc0a75d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev 04534321e450d220d404a240a047c7e1cd82f3fd</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev aaa34e134023e703e197a48d8069cf3a6ebea629</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/228192)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 33955/57048 (59.5%)</li>
<li>Failed: 0/57048 (0.0%)</li>
<li>Not Supported: 23093/57048 (40.5%)</li><li>Warnings: 0/57048 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 33797/57048 (59.2%)</li>
<li>Failed: 0/57048 (0.0%)</li>
<li>Not Supported: 23250/57048 (40.8%)</li><li>Warnings: 1/57048 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 34085/57048 (59.7%)</li>
<li>Failed: 0/57048 (0.0%)</li>
<li>Not Supported: 22963/57048 (40.3%)</li><li>Warnings: 0/57048 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35882/57048 (62.9%)</li>
<li>Failed: 0/57048 (0.0%)</li>
<li>Not Supported: 21166/57048 (37.1%)</li><li>Warnings: 0/57048 (0.0%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — APPROVED** [review]

**Pull request merged**
